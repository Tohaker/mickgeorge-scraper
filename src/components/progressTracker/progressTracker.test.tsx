import React from "react";
import { render } from "@testing-library/react";
import ProgressTracker from ".";

describe("Progress Tracker", () => {
  it("should render with a progress bar and no messages", () => {
    const { getByRole } = render(<ProgressTracker current={20} />);

    const progressBar = getByRole("progressbar");
    const messageBox = getByRole("textbox");

    expect(progressBar).toBeInTheDocument();
    expect(messageBox).toHaveValue("");
  });

  it("should render with a progress bar and some messages", () => {
    const { getByRole } = render(
      <ProgressTracker current={20} message={"here\nis\nsome\nmessage"} />
    );

    const progressBar = getByRole("progressbar");
    const messageBox = getByRole("textbox");

    expect(progressBar).toBeInTheDocument();
    expect(messageBox).toHaveValue("here\nis\nsome\nmessage");
  });
});
