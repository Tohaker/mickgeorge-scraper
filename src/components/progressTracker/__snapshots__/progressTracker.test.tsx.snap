// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Tracker should match snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ iMRMsF"
  >
    <div
      class="progress"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="progress-bar"
        role="progressbar"
        style="width: 0%;"
      >
        0.0%
      </div>
    </div>
    <div
      class="sc-gsTCUz cxqnbh"
    >
      Details:
    </div>
    <textarea
      class="sc-dlfnbm ePegbC"
      readonly=""
    />
    <a
      class="sc-hKgILt eYMeGx"
      disabled=""
      download="export_data.csv"
      href="data:text/csv;charset=utf-8,ï»¿\\"Name\\",\\"Extension\\",\\"Phone Number\\",\\"Application Username\\",\\"Device Type\\",\\"MAC Address\\",\\"Username\\",\\"Status\\",\\"Site name\\",\\"Company Name\\",\\"Voice Recording\\"
\\"firstname surname\\",\\"ext\\",\\"directoryNumber\\",\\"appUserName\\",\\"deviceType\\",\\"macAddress\\",\\"username\\",\\"status\\",\\"site\\",\\"company\\",\\"Voice Recording\\""
      target="_self"
    >
      Export to CSV
    </a>
  </div>
</div>
`;
